js/main.js

```diff
@@ -19,4 +19,76 @@ $form.addEventListener('submit', function (event) {
   data.entries.unshift(formData);
   $img.setAttribute('src', 'images/placeholder-image-square.jpg');
   $form.reset();
+  var $ul = document.querySelector('ul');
+  $ul.prepend(renderEntry(formData));
+  $dataViewEntryForm.className = 'container data-form hidden';
+  $dataViewEntries.className = 'container data-entries';
+  var $entriesDataView = $dataViewEntries.getAttribute('data-view');
+  localStorage.setItem('code-journal-data-view', $entriesDataView);
+  var $noEntriesMessage = document.querySelector('.no-entries');
+  $noEntriesMessage.className = 'no-entries hidden';
 });
+
+function renderEntry(entry) {
+  var $li = document.createElement('li');
+  $li.setAttribute('class', 'row entry');
+  var $img = document.createElement('img');
+  $img.setAttribute('src', entry.photoURL);
+  $img.setAttribute('alt', entry.title);
+  $img.setAttribute('class', 'column-half img-half');
+  $li.appendChild($img);
+  var $div = document.createElement('div');
+  $div.setAttribute('class', 'column-half');
+  $li.appendChild($div);
+  var $h3 = document.createElement('h3');
+  var $title = document.createTextNode(entry.title);
+  $h3.appendChild($title);
+  $div.appendChild($h3);
+  var $p = document.createElement('p');
+  var $notes = document.createTextNode(entry.notes);
+  $p.appendChild($notes);
+  $div.appendChild($p);
+  return $li;
+}
+
+window.addEventListener('DOMContentLoaded', function (event) {
+  var $ul = document.querySelector('ul');
+  for (var i = 0; i < data.entries.length; i++) {
+    $ul.appendChild(renderEntry(data.entries[i]));
+  }
+  var JSONDataView = localStorage.getItem('code-journal-data-view');
+  if (JSONDataView === 'entries') {
+    $dataViewEntryForm.className = 'container data-form hidden';
+    $dataViewEntries.className = 'container data-entries';
+  } else {
+    $dataViewEntryForm.className = 'container data-form';
+    $dataViewEntries.className = 'container data-entries hidden';
+  }
+});
+
+var $nav = document.querySelector('nav');
+var $dataViewEntryForm = document.querySelector('.data-form');
+var $dataViewEntries = document.querySelector('.data-entries');
+$nav.addEventListener('click', function (event) {
+  if (event.target.matches('.entries-link')) {
+    $dataViewEntryForm.className = 'container data-form hidden';
+    $dataViewEntries.className = 'container data-entries';
+  }
+  var $entriesDataView = $dataViewEntries.getAttribute('data-view');
+  localStorage.setItem('code-journal-data-view', $entriesDataView);
+});
+
+var $newButton = document.querySelector('.new-button');
+$newButton.addEventListener('click', function (event) {
+  if (event.target.matches('.new-button')) {
+    $dataViewEntryForm.className = 'container data-form';
+    $dataViewEntries.className = 'container data-entries hidden';
+  }
+  var $formDataView = $dataViewEntryForm.getAttribute('data-view');
+  localStorage.setItem('code-journal-data-view', $formDataView);
```

Comment:
Having this separate localStorage item isn't necessary, you should be updating the value of `data.view` every time the view changes, and then you can use the value of `data.view` when the page loads to determine what the last view was that the user was on.

Reply:
wowwww totally overlooked this property 😅 next time I have to re-read documentation before starting to code!